src/passes/OptimizeInstructions.cpp

```diff
@@ -1153,46 +1153,6 @@ struct OptimizeInstructions
     }
   }
 
-  void visitLocalSet(LocalSet* curr) {
-    // Interactions between local.set/tee and ref.as_non_null can be optimized
-    // in some cases, by removing or moving the ref.as_non_null operation. In
-    // all cases, we only do this when we do *not* allow non-nullable locals. If
-    // we do allow such locals, then (1) this local might be non-nullable, so we
-    // can't remove or move a ref.as_non_null flowing into a local.set/tee, and
-    // (2) even if the local were nullable, if we change things we might prevent
-    // the LocalSubtyping pass from turning it into a non-nullable local later.
-    // Note that we must also check if this local is nullable regardless, as a
-    // parameter might be non-nullable even if nullable locals are disallowed
-    // (as that just affects vars, and not params).
```

Comment:
Why remove this logic? The comment is state, but the code is still useful?

Reply:
Note the `!getModule()->features.hasGCNNLocals()` below. None of this code is supposed to run when non-nullable locals are enabled, so it should all be dead now.

Reply:
Am I reading it wrong? It looks the opposite to me.

When running normally before, all of our users had `!getModule()->features.hasGCNNLocals()` as true (because none of them set that special flag), and so that can be removed from the `&&`-ed conditions in the `if` without changing anything. But if the code is removed, our users could notice.

Reply:
Ugh, yes, I confused myself by forgetting what --enable-gc-nn-locals even did. Ignore all this for now and I'll ping you once I have it cleaned up.